test/suite/test_cursor17.py

```diff
@@ -0,0 +1,96 @@
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+from wiredtiger import stat, WT_NOTFOUND
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_cursor17.py
+# Test optimization that skips page during cursor traversal if it determines that all records on the
+# page have been deleted with a tombstone visible to the current transaction.
+
+class test_cursor17(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=50MB,statistics=(all)'
+    session_config = 'isolation=snapshot'
+
+    def get_stat(self, stat):
+        stat_cursor = self.session.open_cursor('statistics:')
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def test_cursor_skip_pages(self):
+        uri = 'table:test_cursor17'
+        create_params = 'key_format=i,value_format=S'
+        self.session.create(uri, create_params)
+
+        value1 = 'a' * 500
+        total_keys = 40000
+
+        # Keep oldest timestamp pinned.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1))
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(2))
+        cursor = self.session.open_cursor(uri)
+
+        commit_timestamp = 2
+
+        # Insert few thousand key-value pairs.
+        for key in range(total_keys):
+            self.session.begin_transaction()
+            cursor[key] = value1
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(commit_timestamp))
+            commit_timestamp += 1
+
+        # Delete everything on the table except for the first and the last KV pair.
+        for key in range(1, total_keys - 1):
+            self.session.begin_transaction()
+            cursor.set_key(key)
+            self.assertEqual(cursor.remove(),0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(commit_timestamp))
+            commit_timestamp += 1
+
+        # Take a checkpoint to reconcile the pages.
+        self.session.checkpoint()
+
+        self.session.begin_transaction('read_timestamp=' + timestamp_str(commit_timestamp))
+        # Position the cursor on the first record.
+        cursor.set_key(0)
+        self.assertEqual(cursor.search(), 0)
+        # This should move the cursor to the last record.
+        self.assertEqual(cursor.next(), 0)
+        self.assertEqual(cursor.get_key(), total_keys-1)
+
+        # Check if we skipped any pages while moving the cursor.
+        #
+        # I ran this test few times and the stat appeared to be in range of ~1400. I have
```

Comment:
See my first comment - is there a better way to choose the comparison number? Also, please move the explanation for the number chosen to the beginning of the test.

Reply:
I have changed the method for calculating the expected number of skipped pages. Have a look at d5b6ef6.